packages/astro/src/runtime/client/dev-toolbar/ui-library/badge.ts

```diff
@@ -2,21 +2,35 @@ type BadgeSize = 'small' | 'large';
 type BadgeStyle = 'purple' | 'gray' | 'red' | 'green' | 'yellow';
 
 export class DevToolbarBadge extends HTMLElement {
-	size: BadgeSize = 'small';
-	badgeStyle: BadgeStyle = 'purple';
+	_size: BadgeSize = 'small';
+	_badgeStyle: BadgeStyle = 'purple';
+
+	get size() {
+		return this._size;
+	}
+
+	set size(value) {
+		this._size = value;
+		this.updateStyle();
+	}
+
+	get badgeStyle() {
+		return this._badgeStyle;
+	}
+
+	set badgeStyle(value) {
+		this._badgeStyle = value;
+		this.updateStyle();
```

Comment:
Sorry to be super annoying, but what if `badgeStyle` isn't one of the enum values? Can we validate at runtime here? Maybe a fallback to gray is better than an error?

Reply:
Oh no, that's definitely a valid worry to have. I'll add some validation

Reply:
Added validation in the coolest way I know.